scripts/travis/build-all-in-one-image.sh

```diff
@@ -34,18 +42,54 @@ upload_to_docker() {
     echo 'skip docker upload for PR'
     exit 0
   fi
-  export REPO=$1
-  bash ./scripts/travis/upload-to-docker.sh
+
+  REPO=${1}
+  CMD_ROOT=${1}
+
+  unset major minor patch
+
+  if [[ "$BRANCH" == "master" ]]; then
+    TAG="latest"
+  elif [[ $BRANCH =~ ^v([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+    major="${BASH_REMATCH[1]}"
+    minor="${BASH_REMATCH[2]}"
+    patch="${BASH_REMATCH[3]}"
+    TAG=${major}.${minor}.${patch}
+    echo "BRANCH is a release tag: major=$major, minor=$minor, patch=$patch"
+  else
+    TAG="${BRANCH///}"
+  fi
+
+  echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, REPO=$REPO, BRANCH=$BRANCH, TAG=$TAG, IMAGE=$IMAGE"
+
+  IMAGE_TAGS="-t ${REPO}:${TAG}"
+
+  # add major, major.minor and major.minor.patch tags
+  if [[ -n $major ]]; then
+    IMAGE_TAGS="${IMAGE_TAGS} -t ${REPO}:${major}"
+    if [[ -n $minor ]]; then
+       IMAGE_TAGS="${IMAGE_TAGS} -t ${REPO}:${major}.${minor}"
+      if [[ -n $patch ]]; then
+          IMAGE_TAGS="${IMAGE_TAGS} -t ${REPO}:${major}.${minor}.${patch}"
+      fi
+    fi
+  fi
+
+  docker_buildx ${CMD_ROOT} ${IMAGE_TAGS} --platform=linux/amd64,linux/arm64  --push
 }
 
-make build-all-in-one GOOS=linux GOARCH=$GOARCH
+GOOS=linux GOARCH=amd64 make build-all-in-one
+GOOS=linux GOARCH=arm64 make build-all-in-one
```

Comment:
nit: these vars are Make vars, not env vars, they can go on the right side.

Reply:
yep. i reverted it